addons/packages/kapp-controller/0.33.1/bundle/config/values.star

```diff
@@ -0,0 +1,18 @@
+load("@ytt:data", "data")
+
+#export
+values = data.values
+kappNamespace = ""
+if hasattr(values.kappController, 'namespace') and values.kappController.namespace:
+    kappNamespace = values.kappController.namespace
+else:
+    kappNamespace = values.namespace
+end
+
+def generateBashCmdForDNS(coreDNSIP):
+    # This command added the coreDNS IP as the first entry of resolv.conf
+    # In this way, Kapp Controller will have cluster IP access,
+    # and still able to resolve enternal urls when core DNS is unavailable
+
+    return "cp /etc/resolv.conf /etc/resolv.conf.bak; sed '1 i nameserver " + coreDNSIP + "' /etc/resolv.conf.bak > /etc/resolv.conf; rm /etc/resolv.conf.bak; cp -R /etc/* /kapp-etc; chmod g+w /kapp-etc/pki/tls/certs/ca-bundle.crt && chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt"
```

Comment:
I saw some securityContext changes in upstream basefile. Please confirm that those doesn't introduce issues for accessing `/kapp-etc/pki/tls/certs/ca-bundle.crt` file.

`generateBashCmdForDNS ` was added to enable cluster DNS access for kapp-controller. Basically, it uses a init container to edit a volumeMount which has the content of `/etc`, and later attach it to kapp-controller's `/etc`

Reply:
hey @blc1996 What is the context that this starlark code gets run under? We removed the `kapp-controller` user group from our Dockerfile. And so I made a commit here that removes the part where you change the group, but I can't be sure if the command still works without knowing if this is run as `root` or otherwise. I don't think we realized you all were making changes to the ca-crt like this.

Reply:
This code runs as command in initContainer (running as root). It serves the purpose of adding core DNS IP into `/etc/resolv.conf` (only root user can edit) so that kapp-controller can resolve in-cluster service URLs.

We didn't make changes to cert files, the chmod there is just to make sure that kapp-controller container have the write access to the file.

Reply:
gotcha, in that case I think this should still work as is (and removing the `chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt`)

Reply:
Basically, we want to add an entry to `/etc/resolv.conf`, but can't do it within kapp-controller container because it's not running as root.
We added a volumeMount and use a initContainer to copy everything in `/etc` to this volumeMount, and also edit `/etc/resolv.conf` to add the core DNS IP. We then attach this volumeMount to `/etc` of kapp-controller container. Kapp-controller will be read from this volumeMount. Also write to `/etc/pki/tls/certs/ca-bundle.crt`, so we need to make sure it has the right access privilage.

Reply:
Actually I wonder if we need to add `chgrp 2000 /kapp-etc/pki/tls/certs/ca-bundle.crt` back in here? @benmoss 

Let's hold off on merging until I can confirm